requirements/make_base.py

```diff
@@ -0,0 +1,47 @@
+from argparse import ArgumentParser
+
+import tomli
+
+parser = ArgumentParser()
+parser.add_argument(
+    "--nightly",
+    default="",
+    help="List of dependencies to install from main branch for nightly tests, "
+    "separated by commas.",
+)
+args = parser.parse_args()
+
+header = "# Generated by 'tox -e deps', DO NOT EDIT MANUALLY!'\n"
+
+with open("../pyproject.toml", "rb") as toml_file:
+    pyproject = tomli.load(toml_file)
+    dependencies = pyproject["project"].get("dependencies")
+    if not dependencies:
+        raise RuntimeError("No dependencies found in pyproject.toml")
+    dependencies = [dep.strip().strip('"') for dep in dependencies]
+
+with open("base.in", "w") as f:
+    f.write(header)
+    f.write("\n".join(dependencies))
+
+
+def as_nightly(repo: str) -> str:
+    if "/" in repo:
+        org, repo = repo.split("/")
+    else:
+        org = "scipp"
+    if repo == "scipp":
+        version = "cp38"
```

Comment:
I had to hard-code this one since minimum python version of `beamlime` is python3.9.
I opened a related issue: https://github.com/scipp/scipp/issues/3313

Reply:
I suggest to not hardcode anything, just add a 3.9 build, as planned.